src/view/prChangesTreeDataProvider.ts

```diff
@@ -73,6 +73,12 @@ export class PullRequestChangesTreeDataProvider extends vscode.Disposable implem
 		reviewModel: ReviewModel,
 		shouldReveal: boolean,
 	) {
+		if (this._pullRequestManagerMap.has(pullRequestManager)) {
+			const existingNode = this._pullRequestManagerMap.get(pullRequestManager);
+			if (existingNode && (existingNode.pullRequestModel === pullRequestModel)) {
+				return;
+			}
+		}
```

Comment:
Previously, the pull request node would get re-added to the tree view, but the old node wouldn't get disposed of. This led to tree nodes that didn't exist in the tree any more trying to refresh themselves, which in turn let to exceptions that prevented the tree from completing it's refresh.